### DISCUSSION TITLE
Modbus TCP WP steuern (Glen Dimplex System M Cooling)

### DISCUSSION THREAD

**Discussion by Angelfreund** [camueller/SmartApplianceEnabler#207] [Installation und Betrieb]
habe zwei Geräte eingerichtet
1. Zähler für WP, (SMD 72d mit Interfase davor)
2. Schalter (WP SG Ready) als Device-ID 01
Der Zähler funktioniert einwandfrei, Aufruf und Berechnung im Log  ok.
Beim Schalter bekomme ich immer einen Fehler im Log angezeigt.

Jetzt habe ich fogende Info gefunden
Zitat "Noch ein Tipp, damit die WP die Werte annimmt mußt die Register mit dem entsprechenden Wert und FALSE beschreiben. True setzt die WP nach erfolgter Änderung als Bestätigung selbst.
Wenn du den Wert gleich mit TRUE beschreibst wird er nicht gesetzt."

Wer kann helfen ? Was muß ich noch eingeben?
![grafik](https://user-images.githubusercontent.com/66744205/154796322-1a9c824a-1ae8-4d94-8612-a19c259df060.png)

**Comment by camueller** [comment]
Ich würde an Deiner Stelle versuchen, mit einem Modbus-Tool die WP manuell zu schalten. Vermutlich unterstützt Deine WP Modbus/TCP, d.h. Du gehst mit Deinem Modbus-Tool direkt gegen die WP. Falls sie nur Modbus/RTU kann, musst Du gegen den Modbus/TCP-zu-ModbusRTU-Wandler (z.B. `mbusd`) gehen. Erst wenn das klappt und Du die Parameter weisst (Function-Code, Register, Wert), dann kannst Du diese Parameter in den SAE übernehmen.

**Comment by Angelfreund** [comment]
Habe mit dem Modbus Mastertool erfolgreich den Zähler auslesen können, aber nicht die WP.
Von Dimplex habe ich eine neue Anleitung bekommen (alles in englisch :_( ) da blick ich überhaut nicht durch. 
soweit ich das verstanden habe gibt es jetzt die Möglichkeit über MQTT etwas zu machen, geht da irgend etwas mit der 2.0 Version oder hab ich da was falsch verstanden.

**Comment by camueller** [comment]
Lade doch mal die Anleitung irgendwo hoch (Google Drive, Dropbox, ...) und poste den Link hier. Dann schaue ich mir das mal an ...
Danke übrigens für die Spende!!

SAE 2.0 verwendet intern MQTT für die Kommunikation, wodurch sich die Daten gut mit Tools wir Node-RED visualsieren lassen.

**Comment by Angelfreund** [comment]
Der Link zur Beschreibung
https://www.dropbox.com/s/18940no4vjppwkj/GDD-MQTT-Interface-public.pdf?dl=0
und die Register für meinen Sofwarestand sind
Zitat
![grafik](https://user-images.githubusercontent.com/66744205/155499221-14a0d263-f29a-4025-abf3-59674b7d38cb.png)

Integer 166 sollte im Modbus TCP Register 5167 entsprechen. Ich bin mir nicht sicher, ob bei der SystemM im Display der Zustand SG Ready angezeigt wird. Bei „grün“ sollte sich dann die Warmwassersolltemperatur ändern.

**Comment by camueller** [comment]
Die Dokumentation aus dem Link ist ein MQTT-Gateway vor einer Modbus-Schnittstelle. Diese Schnittstelle hilft aktuell nicht weiter, weil der SAE MQTT nur intern verwendet, aber (noch) nicht extern für Zähler/Schalter verwenden kann.

Es muss aber eine Doku direkt für die Modbus-Schnittstelle geben, oder wo hast Du sonst die Registernummern her? Und über diese Schnittstelle sollte es möglich sein, mit dem Modbus Mastertool die WP zu steuern.

**Comment by Angelfreund** [comment]
danke für Deine Bemühungen
Die Daten habe ich vom freundlichen Produktmanagement Systeme von Dimplex erhalten.
Weitere Infos habe ich aus dieser Seite
https://dimplex.atlassian.net/wiki/spaces/DW/pages/2873393288/NWPM+Modbus+TCP#Smart-Grid-/-SG-Ready
meine Software hat aber einen anderen Stand der dort noch nicht dokumentiert ist.
Eigentlich sollte meine WP die Modbus TCP Schnittstelle enthalten, da so bestellt, mal sehen ich forsche weiter.
Du schreibst (noch) nicht, ist dies langfristig vorgesehen? Ich möcht vermeiden noch weitere Programme zu installieren, die ich erst erlernen muß.

**Comment by camueller** [comment]
Aus der verlinkten Anleitung werde ich bzgl. des SG-Ready nicht schlau. Für mich sieht das nicht nach einer Register-Adresses aus, was dort angegeben ist. Wenn Du nicht weiterkommst, würde ich in Erwägung ziehen, den SG-Ready-Kontakt nicht für das Modbus-Interface der WP zu schaltet, sondern extern durch ein Solid State Relais, das Du via GPIO vom Raspberry Pi steuerst.

**Comment by Angelfreund** [comment]
In der neuen Software soll wohl vermieden werden das der SG Ready-Kontakt paralell zum TCP läuft und deshalb die Aussage
![grafik](https://user-images.githubusercontent.com/66744205/157853427-9065e883-74f1-4117-8644-583281f5163d.png)
Ich werde versuchen am WE mal den SG Ready Kontakt manuell zu beleben, ob der überhauft anspricht.
